File: D/575-distribute-candies/main.go
package main

import "fmt"

func distributeCandies(candies []int) int {
	M := map[int]int{}
	for i := 0; i < len(candies); i++ {
		M[candies[i]]++
	}
	r := len(candies) / 2
	if r > len(M) {
		r = len(M)
	}
	return r
}

func main() {
	fmt.Println("a")
}

File: D/575-distribute-candies/main.rs
struct Solution {}

use std::collections::HashSet;
impl Solution {
  pub fn distribute_candies(candy_type: Vec<i32>) -> i32 {
    HashSet::<_>::from_iter(candy_type.iter().copied())
      .len()
      .min(candy_type.len() / 2) as i32
  }
}
